File: leetcode-submissions/1752_Check_if_Array_Is_Sorted_and_Rotated.ts
function is_sorted(nums:number[]):boolean{
   
    for(let i=0;i<nums.length-1;i++){
        if(nums[i]>nums[i+1]){
            return false;
        }
    }
    return true;
}
function check(nums: number[]): boolean {
    let cnt=0;
if(is_sorted(nums)){
    return true;
}
if(nums[0]<nums[nums.length-1]){
    return false;
}
    for(let i=0;i<nums.length-1;i++){
        if(nums[i]>nums[i+1]){
            cnt++;
            if(cnt>1){
                return false;
            }
        }


    }
    return cnt<=1;
};

File: leetcode-submissions/1752_Check_if_Array_Is_Sorted_and_Rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int cnt=0;
        if(nums[0]<nums[nums.size()-1]){
            return  false;
        }
        for(int i=0;i<nums.size()-1;i++){
            if(nums[i]>nums[i+1]){
                cnt++;

            }
            if(cnt>1){
                return false;
            }
            else if(nums[i]==nums[i+1]){
                return false;
            }
        }
        return  cnt<=1;
    }
};